### PR TITLE
feat (sdk): map variables clients on cdsctl

### PR DESCRIPTION
**Pull request by FabienM**

### CODE REVIEW

File: cli/cdsctl/application_variable.go [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+	Args: []cli.Arg{
+		{Name: "project-key"},
+		{Name: "app-name"},
+		{Name: "variable-name"},
+		{Name: "variable-value"},
+	},
+}
+
+func applicationUpdateVariableRun(v cli.Values) error {
+	variable, err := client.ApplicationVariableGet(v["project-key"], v["app-name"], v["variable-name"])
```
Comment (sguiheux) [inline cli/cdsctl/application_variable.go]: You will need old variable name to be able to rename the variable
and so, update name/value and type

File: cli/cdsctl/environment_variable.go [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+	Args: []cli.Arg{
+		{Name: "project-key"},
+		{Name: "env-name"},
+		{Name: "variable-name"},
+		{Name: "variable-value"},
+	},
+}
+
+func environmentUpdateVariableRun(v cli.Values) error {
+	variable, err := client.EnvironmentVariableGet(v["project-key"], v["env-name"], v["variable-name"])
```
Comment (sguiheux) [inline cli/cdsctl/environment_variable.go]: You will need old variable name to be able to rename the variable
update name/value and type

File: cli/cdsctl/project_variable.go [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+	Short: "Update CDS project variable value",
+	Args: []cli.Arg{
+		{Name: "project-key"},
+		{Name: "variable-name"},
+		{Name: "variable-value"},
+	},
+}
+
+func projectUpdateVariableRun(v cli.Values) error {
+	variable, err := client.ProjectVariableGet(v["project-key"], v["variable-name"])
```
Comment (sguiheux) [inline cli/cdsctl/project_variable.go]: You will need old variable name to be able to rename the variable
update name/value and type

File: cli/cdsctl/project_variable.go [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+func projectDeleteVariableRun(v cli.Values) error {
+	return client.ProjectVariableDelete(v["project-key"], v["variable-name"])
+}
+
+var projectVariableUpdateCmd = cli.Command{
+	Name:  "update",
+	Short: "Update CDS project variable value",
+	Args: []cli.Arg{
+		{Name: "project-key"},
+		{Name: "variable-name"},
```
Comment (sguiheux) [inline cli/cdsctl/project_variable.go]: variable type update?

File: sdk/variable.go [hunk 5]
```diff
@@ -4,10 +4,10 @@ import "time"
 
 // Variable represent a variable for a project or pipeline
 type Variable struct {
-	ID    int64  `json:"id"`
-	Name  string `json:"name"`
-	Value string `json:"value"`
-	Type  string `json:"type"`
+	ID    int64  `json:"id,omitempty" cli:"id"`
```
Comment (sguiheux) [inline sdk/variable.go]: id => not write it

File: cli/cdsctl/environment_variable.go [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+}
+
+var environmentVariableUpdateCmd = cli.Command{
+	Name:  "update",
+	Short: "Update CDS environment variable value",
+	Args: []cli.Arg{
+		{Name: "project-key"},
+		{Name: "env-name"},
+		{Name: "variable-name"},
+		{Name: "variable-value"},
```
Comment (sguiheux) [inline cli/cdsctl/environment_variable.go]: variable type + oldname

File: cli/cdsctl/environment_variable.go [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+	},
+}
+
+func environmentCreateVariableRun(v cli.Values) error {
+	variable := &sdk.Variable{
+		Name:  v["variable-name"],
+		Type:  v["variable-type"],
+		Value: v["variable-value"],
+	}
+	if err := client.EnvironmentVariableCreate(v["project-key"], v["env-name"], variable); err != nil {
```
Comment (fsamin) [inline cli/cdsctl/environment_variable.go]: the `if` is useless, let's `return client.EnvironmentVariableCreate`

File: cli/cdsctl/project_variable.go [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+	},
+}
+
+func projectCreateVariableRun(v cli.Values) error {
+	variable := &sdk.Variable{
+		Name:  v["variable-name"],
+		Type:  v["variable-type"],
+		Value: v["variable-value"],
+	}
+	if err := client.ProjectVariableCreate(v["project-key"], variable); err != nil {
```
Comment (fsamin) [inline cli/cdsctl/project_variable.go]: return

**Pull request merged**